File: nan21.dnet.module.hr/nan21.dnet.module.hr.ui.extjs/src/main/resources/webapp/js/ds/PayScaleRateValueDs.js
/* 
 * DNet eBusiness Suite
 * Copyright: Nan21 Electronics SRL. All rights reserved.
 * Use is subject to license terms.
 */

Ext.define("net.nan21.dnet.module.hr.grade.ds.model.PayScaleRateValueDs", {
	extend: 'Ext.data.Model',
	fields: [
		{name:"clientId", type:"int", useNull:true},
		{name:"createdAt", type:"date", dateFormat:Dnet.MODEL_DATE_FORMAT},
		{name:"createdBy", type:"string"},
		{name:"entityFQN", type:"string"},
		{name:"id", type:"int", useNull:true},
		{name:"modifiedAt", type:"date", dateFormat:Dnet.MODEL_DATE_FORMAT},
		{name:"modifiedBy", type:"string"},
		{name:"scalePointCode", type:"string"},
		{name:"scalePointId", type:"int", useNull:true},
		{name:"scaleRateId", type:"int", useNull:true},
		{name:"uuid", type:"string"},
		{name:"validFrom", type:"date", dateFormat:Dnet.MODEL_DATE_FORMAT},
		{name:"validTo", type:"date", dateFormat:Dnet.MODEL_DATE_FORMAT},
		{name:"value", type:"float", useNull:true},
		{name:"version", type:"int", useNull:true}
	]});

Ext.define("net.nan21.dnet.module.hr.grade.ds.model.PayScaleRateValueDsFilter", {
	extend: 'Ext.data.Model',
	fields: [
		{name:"createdAt", type:"date", dateFormat:Dnet.MODEL_DATE_FORMAT},
		{name:"createdAt_From",type:"date", dateFormat:Dnet.MODEL_DATE_FORMAT},
		{name:"createdAt_To",type:"date", dateFormat:Dnet.MODEL_DATE_FORMAT},
		{name:"createdBy", type:"string"},
		{name:"entityFQN", type:"string"},
		{name:"id", type:"int", useNull:true},
		{name:"modifiedAt", type:"date", dateFormat:Dnet.MODEL_DATE_FORMAT},
		{name:"modifiedAt_From",type:"date", dateFormat:Dnet.MODEL_DATE_FORMAT},
		{name:"modifiedAt_To",type:"date", dateFormat:Dnet.MODEL_DATE_FORMAT},
		{name:"modifiedBy", type:"string"},
		{name:"scalePointCode", type:"string"},
		{name:"scalePointId", type:"int", useNull:true},
		{name:"scalePointId_From",type:"int", useNull:true},
		{name:"scalePointId_To",type:"int", useNull:true},
		{name:"scaleRateId", type:"int", useNull:true},
		{name:"scaleRateId_From",type:"int", useNull:true},
		{name:"scaleRateId_To",type:"int", useNull:true},
		{name:"uuid", type:"string"},
		{name:"validFrom", type:"date", dateFormat:Dnet.MODEL_DATE_FORMAT},
		{name:"validFrom_From",type:"date", dateFormat:Dnet.MODEL_DATE_FORMAT},
		{name:"validFrom_To",type:"date", dateFormat:Dnet.MODEL_DATE_FORMAT},
		{name:"validTo", type:"date", dateFormat:Dnet.MODEL_DATE_FORMAT},
		{name:"validTo_From",type:"date", dateFormat:Dnet.MODEL_DATE_FORMAT},
		{name:"validTo_To",type:"date", dateFormat:Dnet.MODEL_DATE_FORMAT},
		{name:"value", type:"float", useNull:true},
		{name:"value_From",type:"float", useNull:true},
		{name:"value_To",type:"float", useNull:true}
	]
});

File: nan21.dnet.module.hr/nan21.dnet.module.hr.presenter/src/main/java/net/nan21/dnet/module/hr/grade/ds/model/PayScaleRateValueDs.java
/* 
 * DNet eBusiness Suite
 * Copyright: 2010-2013 Nan21 Electronics SRL. All rights reserved.
 * Use is subject to license terms.
 */
package net.nan21.dnet.module.hr.grade.ds.model;

import java.util.Date;
import net.nan21.dnet.core.api.annotation.Ds;
import net.nan21.dnet.core.api.annotation.DsField;
import net.nan21.dnet.core.api.annotation.Param;
import net.nan21.dnet.core.api.annotation.RefLookup;
import net.nan21.dnet.core.api.annotation.RefLookups;
import net.nan21.dnet.core.presenter.model.base.AbstractAuditableDs;
import net.nan21.dnet.module.hr.grade.domain.entity.PayScalePoint;
import net.nan21.dnet.module.hr.grade.domain.entity.PayScaleRateValue;

@Ds(entity = PayScaleRateValue.class)
@RefLookups({
		@RefLookup(refId = PayScaleRateValueDs.f_scaleRateId),
		@RefLookup(refId = PayScaleRateValueDs.f_scalePointId, namedQuery = PayScalePoint.NQ_FIND_BY_SCALE_CODE_PRIMITIVE, params = {
				@Param(name = "pPayScaleId", field = PayScaleRateValueDs.f_scaleRateId),
				@Param(name = "pCode", field = PayScaleRateValueDs.f_scalePointCode)})})
public class PayScaleRateValueDs extends AbstractAuditableDs<PayScaleRateValue> {

	public static final String f_scaleRateId = "scaleRateId";
	public static final String f_scalePointId = "scalePointId";
	public static final String f_scalePointCode = "scalePointCode";
	public static final String f_value = "value";
	public static final String f_validFrom = "validFrom";
	public static final String f_validTo = "validTo";

	@DsField(join = "left", path = "scaleRate.id")
	private Long scaleRateId;

	@DsField(join = "left", path = "scalePoint.id")
	private Long scalePointId;

	@DsField(join = "left", path = "scalePoint.code")
	private String scalePointCode;

	@DsField
	private Float value;

	@DsField
	private Date validFrom;

	@DsField
	private Date validTo;

	public PayScaleRateValueDs() {
		super();
	}

	public PayScaleRateValueDs(PayScaleRateValue e) {
		super(e);
	}

	public Long getScaleRateId() {
		return this.scaleRateId;
	}

	public void setScaleRateId(Long scaleRateId) {
		this.scaleRateId = scaleRateId;
	}

	public Long getScalePointId() {
		return this.scalePointId;
	}

	public void setScalePointId(Long scalePointId) {
		this.scalePointId = scalePointId;
	}

	public String getScalePointCode() {
		return this.scalePointCode;
	}

	public void setScalePointCode(String scalePointCode) {
		this.scalePointCode = scalePointCode;
	}

	public Float getValue() {
		return this.value;
	}

	public void setValue(Float value) {
		this.value = value;
	}

	public Date getValidFrom() {
		return this.validFrom;
	}

	public void setValidFrom(Date validFrom) {
		this.validFrom = validFrom;
	}

	public Date getValidTo() {
		return this.validTo;
	}

	public void setValidTo(Date validTo) {
		this.validTo = validTo;
	}
}
